### PR TITLE
Move Checksum Step to Composite

### PR DESCRIPTION
**Pull request by BowTiedDevOps**
Move the step that does checksum generation out of `stacks-core` repository, and change it to use bash instead of a predefined action.

### CODE REVIEW

**Comment by wileyj** [comment]
great! can you link a workflow that uses this action successfully?

**Comment by BowTiedDevOps** [comment]
Yes, here it is: https://github.com/BowTiedDevOps/stacks-core/actions/runs/7920897073/job/21625430204#step:3:57

File: generate-checksum/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+```yaml
+name: Action
+on: push
+jobs:
+  build:
+    name: Job
+    runs-on: ubuntu-latest
+    steps:
+      - name: OpenAPI
+        id: openapi
```
Comment (wileyj) [inline generate-checksum/README.md:24]: probably better to use name/id related to the action being performed, i.e. `sha512sum`

File: generate-checksum/action.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        if ! command -v sha512sum > /dev/null 2>&1; then
+          echo "sha512sum command doesn't exist!";
+          exit 1;
+        fi
+
+        shasum="$(sha512sum release/*)"
+        echo "Generated checksums:"
+        echo "$shasum"
+        echo "$shasum" > "${{ inputs.hashfile_name }}";
+        exit 0;
```
Comment (wileyj) [inline generate-checksum/action.yml:35]: one more error check here might be a good idea. 
simple check that the `shasum` variable isn't empty should suffice. if it is empty, i would exit with non-zero return.

**Review by wileyj — CHANGES_REQUESTED** [review]
overall looks good - a few minor changes please

**Review by wileyj — APPROVED** [review]
👍

**Pull request merged**